junit-jupiter-api/junit-jupiter-api.gradle.kts

```diff
@@ -13,4 +15,20 @@ dependencies {
 	compileOnlyApi(libs.apiguardian)
 
 	compileOnly(kotlin("stdlib"))
+
+	osgiVerification(projects.junitJupiterEngine)
+	osgiVerification(projects.junitPlatformLauncher)
+}
+
+tasks {
+	jar {
+		withConvention(BundleTaskConvention::class) {
+			bnd("""
+				Require-Capability:\
+					org.junit.platform.engine;\
+						filter:='(&(org.junit.platform.engine=junit-jupiter)(version>=${'$'}{version_cleanup;${rootProject.property("version")!!}})(!(version>=${'$'}{versionmask;+;${'$'}{version_cleanup;${rootProject.property("version")!!}}})))';\
+						effective:=active
```

Comment:
Is the intention of this to fail at test runtime when no bundle with this capability, i.e. the Jupiter engine, is not present?

Reply:
Exactly, yet only when resolving (an assembly) using an OSGi resolver. A regular classpath application will ignore all of this. Furthermore, even an OSGi runtime resolver will ignore this since the requirement is not enforceable except during assembly.